src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -85,8 +85,11 @@ export class PtyService extends Disposable implements IPtyService {
 	}
 
 	async attachToProcess(id: number): Promise<void> {
-		this._throwIfNoPty(id);
-		this._logService.trace(`Persistent terminal "${id}": Attach`);
+		try {
+			this._throwIfNoPty(id);
+		} catch (e) {
+			this._logService.trace(`Persistent terminal reconnection failed`, e.message);
+		}
```

Comment:
Would prefer to fail here instead of on `getTerminalLayoutInfo`, we want to restore the layout info if one of the split terminal processes can attach but the other can't

Reply:
I think getTerminalLayoutInfo precedes this call, but have now adjusted so that the error is caught in expandTerminalInstance and thus the valid instances get restored.